devito/finite_differences/finite_difference.py

```diff
@@ -174,7 +176,7 @@ def first_derivative(*args, **kwargs):
     order = int(kwargs.get('order', 1))
     matvec = kwargs.get('matvec', direct)
     side = kwargs.get('side', centered).adjoint(matvec)
-    deriv = 0
+    deriv = Add(0)
```

Comment:
Add(0) makes no sense to me. We could have our own ``Zero`` or ``Number`` (sympy-inherited) class , alternatively

Reply:
Need to initialize it as an `devito.Add` object or falls back to sympy one